internal/alb/ls/listener.go

```diff
@@ -75,6 +75,8 @@ func NewDesiredListener(o *NewDesiredListenerOptions) (*Listener, error) {
 			Priority:         p,
 			Logger:           o.Logger,
 			ListenerRules:    listener.rules,
+			ListenerProtocol: listener.ls.desired.Protocol,
+			ListenerPort:     o.Port,
```

Comment:
minimal comment:
I think it's better be `listener.ls.desired.Port` instead of `o.Port` here, even they are equal.
Which can be understand better 😄, since it's aligned with `listener.ls.desired.Protocol`